app/contribution-hub/ConstraintDetailsModal.tsx

```diff
@@ -0,0 +1,273 @@
+          obj[item] = "";
+          return obj;
+        }, {})
+      );
+    }
+  };
+
+  const [errorMessage, setErrorMessage] = useState<string | null>();
+
+  useEffect(() => {
+    const requirement = requirementList.find(
+      (item) => item.pk == constraint.pk
+    );
+
+    setExistRequirement(requirement ? requirement : null);
+
+    if (requirement) {
+      setIsNotSatisfy(requirement.isNotSatisfy);
+      setRequirementParamsList(requirement.params);
+    } else {
+      createRequirementParamsList();
+    }
+  }, []);
+
+  const checkingParamsValidation = async () => {
+    if (!requirementParamsList) return false;
+    if (
+      !requirementParamsList.COLLECTION_ADDRESS ||
+      !requirementParamsList.CHAIN ||
+      !requirementParamsList.MINIMUM
+    ) {
+      !requirementParamsList.COLLECTION_ADDRESS
+        ? setErrorMessage("Please enter collection address.")
+        : !requirementParamsList.CHAIN
+        ? setErrorMessage("Please select chain.")
+        : setErrorMessage("Please select minimum amount.");
+      return false;
+    }
+
+    if (requirementParamsList.COLLECTION_ADDRESS) {
+      const step2Check = isAddress(requirementParamsList.COLLECTION_ADDRESS);
+      const chain = allChainList?.find(
+        (item) => Number(item.pk) === Number(requirementParamsList.CHAIN)
+      );
+
+      if (!chain) return false;
+
+      const res = await checkCollectionAddress(
+        provider,
+        requirementParamsList.COLLECTION_ADDRESS,
+        Number(chain.chainId)
+      );
+
+      (!step2Check || !res) && setErrorMessage("Invalid contract address.");
+      return step2Check && res;
+    }
+  };
+
+  const checkCsvFileUploadedValidation = () => {
+    console.log(requirementParamsList, requirementParamsList.CSV_FILE);
+    if (!requirementParamsList) return false;
+    if (!requirementParamsList.CSV_FILE) {
+      setErrorMessage("Please upload a csv file.");
+      return false;
+    }
+    return true;
+  };
+
+  const handleAddRequirement = async () => {
+    if (constraint.name === "core.HasNFTVerification") {
+      const res = await checkingParamsValidation();
+      if (!res) return;
+    }
+
+    if (constraint.name === "core.AllowListVerification") {
+      const res = checkCsvFileUploadedValidation();
+      if (!res) return;
+    }
+
+    addRequirements(
+      existRequirement,
+      constraint.pk,
+      constraint.name,
+      constraint.title,
+      isNotSatisfy,
+      requirementParamsList,
+      constraintFile
+    );
+  };
+
+  const handleSelectNotSatisfy = (isSatisfy: boolean) => {
+    setIsNotSatisfy(isSatisfy);
+  };
+
+  return (
+    <div className="flex flex-col gap-2 mt-5 relative">
+      <div
+        className="absolute -top-14 cursor-pointer z-[999]"
+        onClick={handleBackToConstraintListModal}
+      >
+        <Icon
+          iconSrc="/assets/images/provider-dashboard/arrow-left.svg"
+          className="cursor-pointer z-[999999]"
+        />
```

Comment:
@abbasnosrati 

this icon does not have `position: relative` or `position: absolute` applied. remove `z-[999999]`

Reply:
Done!